test/tests/services/fields-flattener.test.js

```diff
@@ -0,0 +1,197 @@
+
+      describe('when a field has not been specified in flatten object', () => {
+        it('should display a warning message', () => {
+          expect.assertions(1);
+
+          jest.resetAllMocks();
+          const fieldsFlattener = new FieldsFlattener({ fields: [] }, [{ }]);
+          fieldsFlattener.validateOptions();
+
+          expect(warnLoggerSpy).toHaveBeenCalledTimes(1);
+        });
+
+        it('should remove the non existing field from fields to flatten', () => {
+          expect.assertions(1);
+
+          const fieldsFlattener = new FieldsFlattener({ fields: [] }, [{ }]);
+          fieldsFlattener.validateOptions();
+
+          expect(fieldsFlattener.flatten).toHaveLength(0);
+        });
+      });
+    });
+  });
+
+  describe('flattening a field', () => {
+    it('should merge fields name with | as separator', () => {
+      expect.assertions(1);
+
+      const schema = generateEngineSchema();
+
+      const fieldsFlattener = new FieldsFlattener(schema, ['engine']);
+      fieldsFlattener.flattenFields();
+
+      expect(schema.fields[0].field).toContain('|');
+    });
+  });
+
+  describe('when the level property is empty', () => {
+    it('should flatten every nested fields until the end', () => {
+      expect.assertions(1);
+
+      const schema = generateEngineSchema();
+      const fieldsFlattener = new FieldsFlattener(schema, ['engine']);
+
+      fieldsFlattener.flattenFields();
+
+      expect(schema.fields).toHaveLength(5);
+    });
+  });
+
+  describe('when the level property is specified', () => {
+    describe('when the level property is above 0', () => {
+      describe('when the level property is higher than actual level', () => {
+        it('should flatten every level until the end', () => {
+          expect.assertions(1);
+
+          const schema = generateEngineSchema();
+          const fieldsFlattener = new FieldsFlattener(schema, [{ field: 'engine', level: 100 }]);
+
+          fieldsFlattener.flattenFields();
+
+          expect(schema.fields).toHaveLength(5);
+        });
+      });
+      describe('when the level property is lower than actual level', () => {
+        it('should flatten only until specified level', () => {
+          expect.assertions(1);
+
+          const schema = generateEngineSchema();
+          const fieldsFlattener = new FieldsFlattener(schema, [{ field: 'engine', level: 0 }]);
+
+          fieldsFlattener.flattenFields();
+
+          expect(schema.fields).toHaveLength(4);
+        });
+      });
+    });
+    describe('when the level property is 0', () => {
+      it('should flatten only direct child field', () => {
+        expect.assertions(1);
+
+        const schema = generateEngineSchema();
+        const fieldsFlattener = new FieldsFlattener(schema, [{ field: 'engine', level: 0 }]);
+
+        fieldsFlattener.flattenFields();
+
+        expect(schema.fields).toHaveLength(4);
+      });
+    });
+    describe('when the level property is under 0', () => {
```

Comment:
what happens if I pass a string or an object to it? for example `level: '0'` seems to work, but `level: 'test'` ignores completely the flattening. I think we should cover those use-cases with tests so the behavior wouldn't change in the future

Reply:
Changed 👍 Now the level is strongly parsed and if the parsing stuff goes wrong, I remove the level and fallback to infinite flattening.